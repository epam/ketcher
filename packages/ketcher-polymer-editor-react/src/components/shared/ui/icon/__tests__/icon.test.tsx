/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

import styled from '@emotion/styled'
import { render, screen } from '@testing-library/react'

import { Icon } from '..'

const StyledIcon = styled(Icon)`
  stroke: green;
`

const MockComponent = () => <StyledIcon name="arrowDown" role="img" />

describe('Icon component', () => {
  it('should render SVG when name is provided', () => {
    render(<MockComponent />)
    const svg = screen.getByRole('img')

    expect(svg).toBeInTheDocument()
  })

  it('should allow styling with emotion/styled', () => {
    render(<MockComponent />)
    const svg = screen.getByRole('img')

    // Making sure some className was set by Emotion/styled
    expect(svg).toHaveAttribute('className')
  })
})
