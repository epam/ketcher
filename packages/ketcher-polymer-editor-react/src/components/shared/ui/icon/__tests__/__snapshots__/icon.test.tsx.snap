// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should render SVG when valid name is provided 1`] = `
<icon-arrow-down.svg
  fill="#525252"
  role="img"
/>
`;

exports[`Icon component should return null when invalid icon name is provided 1`] = `null`;
